metals/src/main/scala/scala/meta/internal/metals/codeactions/CreateCompanionObjectCodeAction.scala

```diff
@@ -68,6 +69,22 @@ class CreateCompanionObjectCodeAction(
 
   }
 
+  private def endPosOfTree(t: Tree): l.Position = {
+    val endMarker: Option[Term.EndMarker] =
+      t.parent.flatMap { p =>
+        val idx = p.children.indexOf(t)
+        // checks if the next child in parent after given tree is `EndMarker`
+        if (idx + 1 < p.children.length) {
+          p.children(idx + 1) match {
+            case marker: Term.EndMarker => Some(marker)
+            case _ => None
+          }
+        } else
+          None
+      }
+    endMarker.map(_.pos).getOrElse(t.pos).toLSP.getEnd
```

Comment:
Because `p.children` is a list I would rather avoid traversing it 3 times (`indexOf`, `p.children.length` and `p.children(idx + 1)`).
```suggestion
    val endMarkerPos: Option[Position] =
      t.parent.flatMap { p =>
        // check if tree just after t is its end marker
        val possibleEndMarker = p.children.dropWhile(_ != t).tail.headOption
        possibleEndMarker match {
          case Some(marker: Term.EndMarker) => Some(marker.pos)
          case _ => None
        }
      }
    endMarkerPos.getOrElse(t.pos).toLSP.getEnd
```

Reply:
thx!